cheatsheets/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -85,24 +91,37 @@ Navigation directives instruct the browser about the locations that the document
 
-| Value           | Description                                                     |
-|-----------------|-----------------------------------------------------------------|
-| 'none'          | No URLs match.                                                  |
-| 'self'          | Refers to the origin site with the same scheme and port number. |
-| 'unsafe-inline' | Allows the usage of inline scripts or styles.                   |
-| 'unsafe-eval'   | Allows the usage of eval in scripts.                            |
+| Value            | Description                                                                                                           |
+|------------------|-----------------------------------------------------------------------------------------------------------------------|
+| 'none'           | No URLs match.                                                                                                        |
+| 'self'           | Refers to the origin site with the same scheme and port number.                                                       |
+| 'unsafe-inline'  | Allows the usage of inline scripts or styles.                                                                         |
+| 'unsafe-eval'    | Allows the usage of eval in scripts.                                                                                  |
+| 'strict-dynamic' | Combining it with hashes or nonces, this tells the browser to trust scripts originating from the root trusted script. |
+
+In case where the developer needs to use inline scripts, it's recommended to use `hashes` for static scripts or a `nonce` on every page request.
 
-In case where the developer needs to use inline scripts, it's recommended to use `sha256` for the script or a `nonce` randomly generated on every page request. 
+To create hashes, check out this [hash generator](https://report-uri.com/home/hash). This is a great [example](https://csp.withgoogle.com/docs/faq.html#static-content) of using hashes.
 
-For more details on hashes and nonces, check out [Scott Helme's Guide](https://scotthelme.co.uk/csp-cheat-sheet/#hashes).
+To better understand how the directive sources work, check out the [source lists from w3c](https://w3c.github.io/webappsec-csp/#framework-directive-source-list).
+
+## Nonces
+
+[Nonces](https://en.wikipedia.org/wiki/Cryptographic_nonce) attributes are added to script tags. Nonce attributes are composed of base64 values. This nonce is verified against the nonce sent in the CSP header, and only matching nonces are allowed to execute.
+
+They can be used in dynamic script blocks in combination with `strict-dynamic`. If the script block is creating additional DOM elements and executing JS inside of them, `strict-dynamic` tells the browser to trust those elements.
+
+_Note:_ `strict-dynamic` is not a standalone directive and should be used in combination with other directive values, such as `https:`, `nonces`, `hashes`, etc.
```

Comment:
The same information about strict-dynamic are doubled here and in the table above (Special Directive Sources)